docs/source/design/github_actions.rst

```diff
@@ -0,0 +1,84 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_github_action:
+
+*******************************
+GitHub Action
+*******************************
+
+.. topic:: Specification scope and status
+
+   This specification describes a proposed interface to a DataLad GitHub Action.
+
+The purpose of the DataLad GitHub Action is to support CI testing with DataLad datasets
+by making it easy to install ``datalad`` and ``get`` data from the dataset.
+
+
+Example Usage
+=============
+
+Dataset installed at ``${GITHUB_WORKSPACE}/studyforrest-data-phase2``,
+``get``'s all the data::
+
+    - uses: datalad/datalad-action@master
+      with:
+        datasets:
+          - source: https://github.com/psychoinformatics-de/studyforrest-data-phase2
+
+Specify advanced options::
+
+    - name: Download testing data
+      uses: datalad/datalad-action@master
+      with:
+        datalad_version: ^0.15.5
+        datasets:
+          - source: https://github.com/psychoinformatics-de/studyforrest-data-phase2
+          - install_path: test_data
+          - recursive: true
+          - recursion_limit: 2
+          - get_paths:
+              - sub-01
+              - sub-02
+              - stimuli
```

Comment:
both `install` and `get` have `--jobs` options. We might want to add here as `jobs` and `get_jobs`.

Also, for the use-case of "get all data", I think we might better just add also `get_data` option to invoke `install` with `--get-data` and avoid individual `get_paths`. Also `get_paths` on itself might be "tricky" to ask for all data -- i.e. `.` would not be recursive.

Reply:
I made a pass at this -- please let me know if I did not follow correctly and more changes are needed!